reduce.js

```diff
@@ -0,0 +1,131 @@
+let arrayOne = [1, 2, 3, 4];
+let arrayTwo = [4, 6, 8];
+const even = element => element % 2 === 0;
+
+const concat = (array, newElement) => {
+    return array.reduce(() => {
+        return [...array, ...newElement];
+    });
+};
+
+console.log(concat(arrayOne, arrayTwo));
+
+const filter = (array, filterFunction) => {
+    return array.reduce(() => {
+        let filterArr = [];
+        for (let elem of array) {
+            if (filterFunction(elem)) {
+                return filterArr.push(array[elem]);
+            }
+        }
+        return undefined;
+    });
+};
+console.log(filter(arrayTwo, even));
+
+const find = (array, func) => {
+    array.reduce(() => {
+        for (let elem of array) {
+            if (func(elem)) {
+                return elem;
+            }
+        }
+        return undefined;
+    });
+};
+
+console.log(find(arrayOne, even));
+
+const findIndex = (array, condition) => {
```

Comment:
you need to implement these methods using reduce, but not loops
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce

please read how reduce works